.buildkite/steps/vault-init-tester.sh

```diff
@@ -1,16 +1,19 @@
 #!/bin/bash
 
-# This token is defined in docker-compoise.yml and is for testing ONLY!!!!!!
+# This token is defined in docker-compose.yml and is for testing ONLY!!!!!!
 AUTH_TOKEN="${BUILDKITE_PLUGIN_VAULT_SECRETS_AUTH_TOKEN:-}"
 PROJECT=${BUILDKITE_PIPELINE_SLUG:-foobar_project}
-vault auth - <<< "$AUTH_TOKEN"
+export VAULT_ADDR="http://vault-svc:8200"
 
-[ $? -eq 0 ] && vault mount -path data/buildkite kv
 
-TESTDATA_1=`echo TESTDATA_1=\"foo bar 1\" | base64`
-TESTDATA_2=`echo TESTDATA_2=\"foo bar 2\" | base64`
+vault login token="$AUTH_TOKEN"
+
+[ $? -eq 0 ] && vault secrets enable -path=data/buildkite kv
+
+TESTDATA_1="foobar1"
+TESTDATA_2="foobar2"
 
 [ $? -ne 1 ] && {
```

Comment:
@toote I think this is valid since it's not a variable assignment but rather an operation on the vault server. The last command run is `vault login token="$AUTH_TOKEN"` which will authenticate to the vault service that is spun up as part of the integration tests using the `AUTH_TOKEN`, it tests that the script is able to login to `vault` with the supplied token and enable the `data/buildkite` kv2 secrets engine. There might be a more efficient way of doing this though.

Reply:
What I'm saying is that the value of `$?` (the exit code of the command in line 9) gets overwritten on every single command run. In particular this is the value of the variable for each line:

* line 11: if `vault login` failed or not
* line 13: 
  - if it was `0` in line 11, it is now the result of the `vault secrets enable`
  - otherwise, it will remain unchanged
* line 14: `0` (because the variable assignment of line 13 can not fail)
* live 16: `0` (because the variable assignment of line 14 can not fail)

I double-checked this:
```
$ bash -c 'exit 1'
$ echo $?
1
$ bash -c 'exit 1'
$ [ $? -eq 1 ] && echo true
true
$ echo $?
0
$ bash -c 'exit 1'
$ [ $? -eq 0 ] && echo true
$ echo $?
1
$ bash -c 'exit 1'
$ FOO=10
$ echo $?
0
```

Based on what I believe it is attempted here, and avoiding the use and abuse of `$?` and `test` in the form of `[]`, the script should be something like:

```
if ! vault login ...; then
  echo 'was not able to log in'
  exit 1
fi

if ! vault mount -path data/buildkite kv; then
  echo 'was not able to mount'
  exit 2 # just to be different from the login error
fi

vault kv put data..
```

Reply:
Ahhh, this is where my knowledge of bash is pretty weak so I've learned some new things here. I've tested your suggestions and things are working as expected, thanks for that explanation!